classes/mix-rustler.bbclass

```diff
@@ -0,0 +1,51 @@
+inherit mix-rebar3 cargo_common
+
+# We don't want to use bitbakes vendoring
+CARGO_DISABLE_BITBAKE_VENDORING = "1"
+
+# the binary we will use
+CARGO = "cargo"
+
+# We need cargo to compile for the target
+BASEDEPENDS:append = " cargo-native"
+
+# Ensure we get the right rust variant
+DEPENDS:append:class-target = " virtual/${TARGET_PREFIX}rust ${RUSTLIB_DEP}"
+DEPENDS:append:class-nativesdk = " virtual/${TARGET_PREFIX}rust ${RUSTLIB_DEP}"
+DEPENDS:append:class-native = " rust-native"
+
+# Enable build separation
+#B = "${WORKDIR}/build"
+
+# In case something fails in the build process, give a bit more feedback on
+# where the issue occured
+export RUST_BACKTRACE = "1"
+
+# The directory of the Cargo.toml relative to the root directory, per default
+# assume there's a Cargo.toml directly in the root directory
+CARGO_SRC_DIR ??= ""
+
+# The actual path to the Cargo.toml
+MANIFEST_PATH ??= "${S}/${CARGO_SRC_DIR}/Cargo.toml"
+
+RUSTFLAGS ??= ""
+BUILD_MODE = "${@['--release', ''][d.getVar('DEBUG_BUILD') == '1']}"
+CARGO_BUILD_FLAGS = "-v --target ${HOST_SYS} ${BUILD_MODE} --manifest-path=${MANIFEST_PATH}"
+
+# This is based on the content of CARGO_BUILD_FLAGS and generally will need to
+# change if CARGO_BUILD_FLAGS changes.
+BUILD_DIR = "${@['release', 'debug'][d.getVar('DEBUG_BUILD') == '1']}"
+CARGO_TARGET_SUBDIR="${HOST_SYS}/${BUILD_DIR}"
+
+do_configure:append() {
+    mix_do_configure
+}
+
+do_configure:prepend() {
+    oe_cargo_fix_env
+    export RUSTFLAGS="${RUSTFLAGS}"
+	export RUST_TARGET_PATH="${RUST_TARGET_PATH}"
+	bbnote "cargo = $(which ${CARGO})"
+	bbnote "rustc = $(which ${RUSTC})"
+	bbnote "${CARGO} build ${CARGO_BUILD_FLAGS} $@"
+}
```

Comment:
I appended this block here, so this can be used by rustler when calling cargo.

```
do_compile:prepend() {
    export HOST_SYS="${RUST_HOST_SYS}"
}
```
To use HOST_SYS, i added this: https://github.com/chrta/meta-erlang-hello-world/commit/b36cefbdbbf456a1256f19f0c89e85091ebd1e4c

Now cargo is called with the correct target, but dependencies are missing:
```
ERROR: hello-elixir-rustler-0.1.0-gitAUTOINC+b36cefbdbb-r0 do_compile: ExecutionError('/opt/yocto/build_lbel/tmp/work/cortexa7t2hf-neon-fslc-linux-gnueabi/hello-elixir-rustler/0.1.0-gitAUTOINC+b36cefbdbb-r0/temp/run.do_compile.9564', 1, None, None)
ERROR: Logfile of failure stored in: /opt/yocto/build_lbel/tmp/work/cortexa7t2hf-neon-fslc-linux-gnueabi/hello-elixir-rustler/0.1.0-gitAUTOINC+b36cefbdbb-r0/temp/log.do_compile.9564
Log data follows:
| DEBUG: Executing shell function do_compile
| Compiling 2 files (.ex)
| Compiling crate helloelixirrustler_rustlernif in release mode (native/helloelixirrustler_rustlernif)
|    Compiling void v1.0.2
|    Compiling lazy_static v1.4.0
|    Compiling rustler v0.25.0
|    Compiling proc-macro2 v1.0.43
|    Compiling aho-corasick v0.7.18
|     Building [===========>               ] 13/28: void, rustler(build), laz...
|     Building [============>              ] 14/28: void, lazy_static, proc-m...
| warning: target json file contains unused fields: has-elf-tls
| 
| error[E0463]: can't find crate for `core`
|   |
|   = note: the `arm-unknown-linux-gnueabihf` target may not be installed
|   = help: consider downloading the target with `rustup target add arm-unknown-linux-gnueabihf`
| 
| error[E0463]: can't find crate for `compiler_builtins`
| 
| error[E0463]: can't find crate for `core`
|   --> /opt/yocto/build_lbel/tmp/work/cortexa7t2hf-neon-fslc-linux-gnueabi/hello-elixir-rustler/0.1.0-gitAUTOINC+b36cefbdbb-r0/cargo_home/registry/src/github.com-1ecc6299db9ec823/void-1.0.2/src/lib.rs:16:5
|    |
| 16 |     extern crate core;
|    |     ^^^^^^^^^^^^^^^^^^ can't find crate
|    |
|    = note: the `arm-unknown-linux-gnueabihf` target may not be installed
|    = help: consider downloading the target with `rustup target add arm-unknown-linux-gnueabihf`
```